contracts/DVA/DVATransferManager.sol

```diff
@@ -0,0 +1,651 @@
+    *  the event is emitted by 'rejectTransfer' function.
+    *  `transferID` is the unique ID of the transfer
+    *  `rejectedBy` is the approver address
+    */
+    event TransferRejected(
+        bytes32 transferID,
+        address rejectedBy
+    );
+
+    /**
+    *  this event is emitted whenever a transfer is cancelled by the sender
+    *  the event is emitted by 'cancelTransfer' function.
+    *  `transferID` is the unique ID of the transfer
+    */
+    event TransferCancelled(
+        bytes32 transferID
+    );
+
+    /**
+    *  this event is emitted whenever all approvers approve a transfer
+    *  the event is emitted by 'approveTransfer' function.
+    *  `transferID` is the unique ID of the transfer
+    *  `tokenAddress` is the token address
+    *  `sender` is the address of the sender
+    *  `recipient` is the address of the recipient
+    *  `amount` is the amount of the transfer
+    */
+    event TransferCompleted(
+        bytes32 transferID,
+        address tokenAddress,
+        address sender,
+        address recipient,
+        uint256 amount
+    );
+
+    /**
+     *  this event is emitted whenever a transfer approval criteria are reset
+     *  the event is emitted by 'approveTransfer' and 'rejectTransfer' functions.
+     *  `transferID` is the unique ID of the transfer
+     *  `approvers` is the list of approvers
+     *  `approvalCriteriaHash` is the approval criteria hash
+     */
+    event TransferApprovalStateReset(
+        bytes32 transferID,
+        Approver[] approvers,
+        bytes32 approvalCriteriaHash
+    );
+
+    error OnlyTokenAgentCanCall(address _tokenAddress);
+
+    error OnlyTransferSenderCanCall(bytes32 _transferID);
+
+    error TokenIsNotRegistered(address _tokenAddress);
+
+    error RecipientIsNotVerified(address _tokenAddress, address _recipient);
+
+    error DVAManagerIsNotVerifiedForTheToken(address _tokenAddress);
+
+    error TokenTransferFailed(address _tokenAddress, address _from, address _to, uint256 _amount);
+
+    error InvalidTransferID(bytes32 _transferID);
+
+    error TransferIsNotInPendingStatus(bytes32 _transferID);
+
+    error ApprovalsMustBeSequential(bytes32 _transferID);
+
+    error ApproverNotFound(bytes32 _transferID, address _approver);
+
+    error SignaturesCanNotBeEmpty(bytes32 _transferID);
+
+    constructor(){
+        _txNonce = 0;
+    }
+
+    /**
+    *  @dev modify the approval criteria of a token
+     *  @param tokenAddress is the token address.
+     *  @param includeRecipientApprover determines whether the recipient is included in the approver list
+     *  @param includeAgentApprover determines whether the agent is included in the approver list
+     *  @param sequentialApproval determines whether approvals must be sequential
+     *  @param additionalApprovers are the addresses of additional approvers to be added to the approver list
+     *  Only an agent of a token can call this function
```

Comment:
Agents on T-REX are in charge of operational tasks such as minting, burning, freezing, recovering lost tokens, etc. Here i believe the task is more related to the general transfer rules management for the token, which is under the owner scope in T-REX, in the same way that it is the token owner that specifies the compliance rules, or the eligibility rules, i believe it should be the Token owner specifying the approval criterias in DvA